core/src/test/java/org/apache/iceberg/rest/requests/TestCommitTransactionRequestParser.java

```diff
@@ -0,0 +1,245 @@
+        .hasMessage("Cannot parse to a string value: name: 23");
+  }
+
+  @Test
+  public void invalidRequirements() {
+    assertThatThrownBy(
+            () ->
+                CommitTransactionRequestParser.fromJson(
+                    "{\"table-changes\":[{\"identifier\":{\"namespace\":[\"ns1\"],\"name\":\"table1\"}}]}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse missing field: requirements");
+
+    assertThatThrownBy(
+            () ->
+                CommitTransactionRequestParser.fromJson(
+                    "{\"table-changes\":[{\"identifier\":{\"namespace\":[\"ns1\"],\"name\":\"table1\"},"
+                        + "\"requirements\":[23],\"updates\":[]}]}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse update requirement from non-object value: 23");
+
+    assertThatThrownBy(
+            () ->
+                CommitTransactionRequestParser.fromJson(
+                    "{\"table-changes\":[{\"identifier\":{\"namespace\":[\"ns1\"],\"name\":\"table1\"},"
+                        + "\"requirements\":[{}],\"updates\":[]}]}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse update requirement. Missing field: type");
+
+    assertThatThrownBy(
+            () ->
+                CommitTransactionRequestParser.fromJson(
+                    "{\"table-changes\":[{\"identifier\":{\"namespace\":[\"ns1\"],\"name\":\"table1\"},"
+                        + "\"requirements\":[{\"type\":\"assert-table-uuid\"}],\"updates\":[]}]}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse missing string: uuid");
+  }
+
+  @Test
+  public void invalidMetadataUpdates() {
+    assertThatThrownBy(
+            () ->
+                CommitTransactionRequestParser.fromJson(
+                    "{\"table-changes\":[{\"identifier\":{\"namespace\":[\"ns1\"],\"name\":\"table1\"},\"requirements\":[]}]}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse missing field: updates");
+
+    assertThatThrownBy(
+            () ->
+                CommitTransactionRequestParser.fromJson(
+                    "{\"table-changes\":[{\"identifier\":{\"namespace\":[\"ns1\"],\"name\":\"table1\"},"
+                        + "\"requirements\":[],\"updates\":[23]}]}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse metadata update from non-object value: 23");
+
+    assertThatThrownBy(
+            () ->
+                CommitTransactionRequestParser.fromJson(
+                    "{\"table-changes\":[{\"identifier\":{\"namespace\":[\"ns1\"],\"name\":\"table1\"},"
+                        + "\"requirements\":[],\"updates\":[{}]}]}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse metadata update. Missing field: action");
```

Comment:
I think this is more of a validation for `UpdateTableRequest`.

Reply:
yes you're right, since we initially didn't have a parser for UpdateTableRequest, I added them here.

I've removed the tests for invalid updates/requirements now and in https://github.com/apache/iceberg/pull/7569 we'll have those tests at the proper place